173 Lighting tweaks and intergration

TheDarkElites opened the pull request:
## About the Pull Request

Makes it so that 173 can now break lights, and the light level of the tile that 173 is on now affects the blink cooldown.

## Why It's Good For The Game

Adds a bit more gameplay to 173 besides waiting, trying to get near a person, and left-clicking. In full darkness, 173 can move un-impeded even if he is "being looked at" meaning 173 players can try to be more tactical about their approach and take out lights in areas they might come in contact with people. 173 players can also now try to make a getaway in certain situations. For example, if four people are staring it down, but it manages to break a nearby light, it can still escape. On the other hand, security now also has to worry about keeping flashlights around, and power outages become a serious threat if 173 is breached. Finally, the power outage that happens in 173's chamber during a breach now actually impacts the situation instead of just being an aesthetic choice.

P.S.: Im aware this is somewhat of a buff but because it's in the form of a new gameplay mechanic and breaching takes an HOUR, that's right, a full HOUR, I think it's fair.

## Changelog

:cl:
add: 173 can now break lights
add: Light level now affects blink cooldown
/:cl:

TheDarkElites committed:
Light System Implementation

- 173 can now break lights.
- blink time is now affected by the light level of the tile on which 173 resides.

TheDarkElites committed:
Made 173 unable to break lights in his own CZ

-173 cant break lights in his own containment zone

TheDarkElites committed:
Reverted snap cooldown

Snap cool down was lowered to two seconds while testing, reverted to the original four second cooldown.

TheDarkElites committed:
Fixed blink_cooldown not working

Sometimes I put the parenthesis in the wrong spot :(

aidan-bowman commented:
Good pr, gives engineers and janitor more importance and also makes more sense with how 173 should work.

Although, you should probably use physical_attack_hand() instead of creating a new function, for future compatibility.

Also, it's probably be nice to expand this to other eyesight-based skips (e.g. 096).

TheDarkElites committed:
Cleanup

Replaced floor_lights custom shatter proc with already created physical_attack_hand() proc for overall cleanness. (Thanks cheesePizza2!)
